src/vulkan/vulkan_types.h

```diff
@@ -58,15 +58,19 @@ namespace IC {
     };
 
     struct PointLightDescriptors {
-        glm::vec3 pos;
-        glm::float32 ambientStrength;
-        glm::vec3 color;
-        glm::float32 padding;
+        alignas(16) glm::vec3 pos;
```

Comment:
What does `alignas` do and what's the benefit of changing the type here?

Reply:
There are alignment rules for data passed to shaders. 
```
    Scalars have to be aligned by N (= 4 bytes given 32 bit floats).
    A vec2 must be aligned by 2N (= 8 bytes)
    A vec3 or vec4 must be aligned by 4N (= 16 bytes)
    A nested structure must be aligned by the base alignment of its members rounded up to a multiple of 16.
    A mat4 matrix must have the same alignment as a vec4.
```
vec3 has to be aligned to 16 bytes, and since a vec3 is 12 bytes (3 floats * 4 bytes/float) I have to manually specify the alignment. This was absent in the previous code because I put a single float in between. It's probably cleaner to specify alignment like this anyway though. 

Not sure what you mean by changing the type? It's still a vec3. Unless you mean the ambient stuff, that's changed to give more control over the ambient lighting color.